### ISSUE TITLE
Running program.populate during training causes loss to be None for PrimalDual

### ISSUE THREAD

**Issue by AlexWan0** [HLR/DomiKnowS#342]
Issue: Using program.populate in the middle of a training iteration results in the loss being None on the next training epoch.

Seems to be because program.populate sets the mode as Mode.POPULATE. The model should set it back on the beginning of training, but PrimalDual program uses self.train(), which only changes the pytorch setting.

Changing line 100 of primaldualprogram.py to self.model.mode(Mode.TRAIN) fixes this.

**Comment by hfaghihi15** [comment]
Hi @AlexWan0 could you please check and see if this issue is resolved and close it if so!
